internal/cmd/skupper/token/kube/token_redeem.go

```diff
@@ -0,0 +1,187 @@
+
+	// Validate token file name
+	if len(args) < 1 {
+		validationErrors = append(validationErrors, fmt.Errorf("token file name must be configured"))
+	} else if len(args) > 1 {
+		validationErrors = append(validationErrors, fmt.Errorf("only one argument is allowed for this command"))
+	} else if args[0] == "" {
+		validationErrors = append(validationErrors, fmt.Errorf("file name must not be empty"))
+	} else {
+		ok, err := tokenStringValidator.Evaluate(args[0])
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("token file name is not valid: %s", err))
+		} else {
+			cmd.fileName = args[0]
+		}
+	}
+
+	// Validate there is already a site defined in the namespace before a token can be redeemed
+	siteList, _ := cmd.client.Sites(cmd.namespace).List(context.TODO(), metav1.ListOptions{})
+	if siteList == nil || len(siteList.Items) == 0 {
+		validationErrors = append(validationErrors, fmt.Errorf("A site must exist in namespace %s before a token can be redeemed", cmd.namespace))
+	} else {
+		for _, s := range siteList.Items {
+			if s.Status.Status.StatusMessage == "OK" && s.Status.Active {
+				cmd.activeSite = &s
+			}
+		}
+		if cmd.activeSite == nil {
+			validationErrors = append(validationErrors, fmt.Errorf("there is no active skupper site in this namespace"))
+		}
+	}
+
+	// Validate if token file exists
+	if cmd.fileName != "" {
+		_, err := os.Stat(cmd.fileName)
+		if err != nil {
+			validationErrors = append(validationErrors, fmt.Errorf("token file does not exist: %s", err))
+		}
+	}
+
+	//TBD what is valid times
+	if cmd.flags.timeout <= 0*time.Minute {
+		validationErrors = append(validationErrors, fmt.Errorf("timeout is not valid"))
+	}
+
+	return validationErrors
+}
+
+func (cmd *CmdTokenRedeem) Run() error {
+
+	// get data from token file
+	var grant v1alpha1.AccessGrant
+	var tokenFile []byte
+	tokenFile, err := os.ReadFile(cmd.fileName)
+	if err != nil {
+		err = fmt.Errorf("unable to read token file - %v", err)
+		return err
+	}
+	yamlS := json.NewSerializerWithOptions(json.DefaultMetaFactory, scheme.Scheme, scheme.Scheme, json.SerializerOptions{Yaml: true})
+	yamlS.Decode(tokenFile, nil, &grant)
+
+	// create AccessToken
+	resource := v1alpha1.AccessToken{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "skupper.io/v1alpha1",
+			Kind:       "AccessToken",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      grant.Name,
+			Namespace: cmd.namespace,
+		},
+		Spec: v1alpha1.AccessTokenSpec{
+			Url:  grant.Status.Url,
+			Ca:   grant.Status.Ca,
+			Code: grant.Status.Code,
+		},
+	}
+
+	_, err = cmd.client.AccessTokens(cmd.namespace).Create(context.TODO(), &resource, metav1.CreateOptions{})
+	return err
+}
+
+func (cmd *CmdTokenRedeem) WaitUntilReady() error {
+
+	err := utils.NewSpinnerWithTimeout("Waiting for token status ...", int(cmd.flags.timeout.Seconds()), func() error {
+
+		token, err := cmd.client.AccessTokens(cmd.namespace).Get(context.TODO(), cmd.name, metav1.GetOptions{})
```

Comment:
`cmd.name` is always empty here

Reply:
I must have removed that accidentally when cleaning up code.   Fixed